src/apps/weblib/js-api/events.js

```diff
@@ -45,24 +51,27 @@ export default class Events {
     })
 
     if (handlers.length === 0) {
-      const cname = this.vizzu._toCString(eventName)
-      try {
-        this.vizzu._call(this.module._removeEventListener)(cname, cfunc)
-      } finally {
-        this.module._free(cname)
+      if (!this._isJSEvent(eventName)) {
+        const cname = this.vizzu._toCString(eventName)
+        try {
+          this.vizzu._call(this.module._removeEventListener)(cname, cfunc)
+        } finally {
+          this.module._free(cname)
+        }
+        this.module.removeFunction(cfunc)
       }
-      this.module.removeFunction(cfunc)
       this.eventHandlers.delete(eventName)
     }
   }
 
   _invoke(eventName, param) {
+    let canceled = false
     try {
       if (this.eventHandlers.has(eventName)) {
-        const jsparam = this.vizzu._fromCString(param)
-
-        for (const handler of [...this.eventHandlers.get(eventName)[1]]) {
-          const eventParam = JSON.parse(jsparam)
+        let eventParam
+        if (!this._isJSEvent(eventName)) {
+          const jsparam = this.vizzu._fromCString(param)
+          eventParam = JSON.parse(jsparam)
           eventParam.preventDefault = () => {
             this.vizzu._call(this.module._event_preventDefault)()
           }
```

Comment:
Can you move this "c" eventParam creation to a helper function?

Reply:
yes